### ISSUE TITLE
Crash when using MFM with mesh with Z size 2, 3, 4 or 5

### ISSUE THREAD

**Issue by jsampaio** [mumax/3#93]
When the number of layers (mesh z size) is 2 or 4, mumax crashes when calculating the MFM kernel.  The error message comes from line 38 of mfmkernel.go :
```
36 util.AssertMsg(size[X]%2 == 0 && size[Y]%2 == 0, "Even kernel size needed")
37		if size[Z] > 1 {
38			util.AssertMsg(size[Z]%2 == 0, "Even kernel size needed")
39		}
```
Surprisingly (for me...), if you set the mesh size in X or Y to an odd number (127) it calculates the MFM kernel with no problems.

Here's the log of an interactive session, where I set the mesh Z size to 2 and selected the MFM image:

```
SetMesh(128, 64, 2, 4e-09, 4e-09, 4e-09, 0, 0, 0)
//resizing...
// Initializing geometry 0 %
// Initializing geometry 100 %
calculating MFM kernel
Even kernel size needed
panic: Even kernel size needed
[...]
	/home/joao/go/src/github.com/mumax/3/mag/mfmkernel.go:38 +0x631
github.com/mumax/3/cuda.NewMFM(0x1117e20, 0x3e6ad7f29abcaf48, 0x3f50624dd2f1a9fc, 0xc82115f880)
[...]
```

**Comment by jsampaio** [comment]
This bug still happens, I can't follow the MFM code to correct it... if anyone wants to try, here's a mx3 script that triggers it:

```
Msat = 1e6; Aex = 10e-12;
nZ := 2;
SetMesh(128, 128, nZ, 4e-09, 4e-09, 4e-09, 0, 0, 0);
m  = vortex(1, 1);
print("Trying the MFM kernel for nZ=",nZ);
MFMLift = 40e-9; //forces the calculation of the MFM kernel.
print("survived nZ=",nZ);
```

this works for nZ=1, but crashes for nZ=2, 3, 4 or 5.

**Comment by JonathanMaes** [comment]
This crash can be avoided by commenting out `mag/mfmkernel.go:113`:
```go
util.AssertMsg(size[Z]%2 == 0, "Even kernel size needed")
```
**The question is why this assert exists in the first place?** Without this assert, all tests still run as expected, and the resulting MFM images (generated by adding `Snapshot(MFM)` at the end of the previous comment's example script) seem just fine for `nZ <= 5`.

This bug happens because, at the start of the function the assert is in, the mesh is resized using `padSize()`. But `padSize` has an `if` statement which causes the z-axis to be padded to an odd length `2*N - 1` if `N<=SMALL_N` (with `SMALL_N=5`), and otherwise padded to even length `2*N`. Therefore, the assert which checks if the axis has an even length fails if `nZ <= SMALL_N = 5`.

**Comment by JLeliaert** [comment]
Have you checked if it returns correct results with PBC's as well for odd sizes?

**Comment by JonathanMaes** [comment]
@JLeliaert I have checked with odd `Nx`, `Ny` and all `Nz<=10`, and tested on my old GPU with a binary compiled for CUDA v10.0 (the lowest we will support from 3.11 onwards) and CC6.1. This gave the same result as on a system with CUDA v12.6 and CC8.6. If the reason for the even kernel size requirement was CUDA-related, it seems to have been addressed before v10.0.

**Comment by JonathanMaes** [comment]
The odd-size assert checks were removed in 3.11. If any new error would arise due to this removal, a new issue can be opened.
